app/main.py

```diff
@@ -1,2 +1,123 @@
+            if (self.storage[new_index] is None
+                    or self.storage[new_index][0] == key):
+                return new_index
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        """
+        This method is responsible for placing
+        the key-value-hash tuple into storage.
+        """
+        self.size_storage()
+        hash_, index_ = self.get_index_hash(key)
+
+        if self.storage[index_] is None:
+            self.storage[index_] = (key, value, hash_)
+            self.current_length += 1
+        else:
+            if self.storage[index_][0] == key:
+                self.storage[index_] = (key, value, hash_)
+            else:
+                new_index = self.collision(key, index_)
+                if self.storage[new_index] is None:
+                    self.storage[new_index] = (key, value, hash_)
+                    self.current_length += 1
+                else:
+                    self.storage[new_index] = (key, value, hash_)
+
+    def __getitem__(self, key: Any) -> Any:
+        """This method is responsible for getting the value by key."""
+        hash_, index_ = self.get_index_hash(key)
+
+        if self.storage[index_] is None:
+            raise KeyError("No such key exists.")
+
+        if (self.storage[index_][0] == key
+                and self.storage[index_][2] == hash_):
+            return self.storage[index_][1]
+
+        for i in range(1, self.capacity):
+            new_index = (index_ + i) % self.capacity
+            if self.storage[new_index] is None:
+                raise KeyError("No such key exists.")
+            if (self.storage[new_index][0] == key
+                    and self.storage[new_index][2] == hash_):
+                return self.storage[new_index][1]
+
+    def clear(self) -> None:
+        """This method returns the dictionary to its original form."""
+        self.capacity = 8
+        self.storage = [None] * self.capacity
+        self.current_length = 0
+
+    def __delitem__(self, key: Any) -> None:
+        """This method removes an element by the specified key."""
+        hash_, index_ = self.get_index_hash(key)
+
+        if self.storage[index_] is None:
+            raise KeyError("No such key exists.")
+
+        if self.storage[index_][0] == key and self.storage[index_][2] == hash_:
+            self.storage[index_] = None
+            self.current_length -= 1
+            return f"The item {key} has been successfully removed."
+
+        for i in range(1, self.capacity):
+            new_index = (index_ + i) % self.capacity
+            if self.storage[new_index] is None:
+                raise KeyError("No such key exists.")
+
+            if (self.storage[new_index][0] == key
+                    and self.storage[new_index][2] == hash_):
+                self.storage[new_index] = None
+                self.current_length -= 1
+                return f"The item {key} has been successfully removed."
```

Comment:
Returning a string message upon successful deletion is not typical for dictionary-like objects in Python. Consider removing the return statement or returning `None` instead.